packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -1976,12 +1926,22 @@ async function resolveIncludeFileUri(
 
   if (pgroup) {
     // lib file as either a string or a member from a known process group
+    const absPathRegex = /^\/|\\|[A-Z]:/i;
     for (const lib of pgroup.libs) {
-      const libFileUri = UriUtils.joinPath(
-        URI.parse(PluginConfigurationProviderInstance.getWorkspacePath()),
-        lib,
-        item.fileName,
-      );
+      let libFileUri: URI;
+      if (!absPathRegex.test(lib)) {
+        // relative lib path, combine w/ workspace
+        libFileUri = UriUtils.joinPath(
+          URI.parse(PluginConfigurationProviderInstance.getWorkspacePath()),
+          lib,
+          item.fileName,
+        );
+      } else {
+        // use lib path over workspace
+        const libUri = URI.file(lib);
```

Comment:
Minor: We should support non-file schemes here as well. This might happen in the web extension for example.

```suggestion
        const libUri = URI.file(lib).with({
          scheme: context.entryUri.scheme
        });
```

Reply:
I'm assuming the entryUri will always be set when we're in a compile unit? I can see it's optional and I'm curious about that.

Reply:
Feel free to remove the `?` from the `entryUri` and `currentUri` fields. I probably made those optional under some old assumptions that no longer hold up.

Note that the function will already have returned at this point if the `entryUri` is not set.

Reply:
Gotcha, I'll tweak those declarations as well before pulling this in.